Code/Legacy/CrySystem/System.cpp

```diff
@@ -289,6 +290,30 @@ void CSystem::ShutDown()
 {
     CryLogAlways("System Shutdown");
 
+    // Disconnect any networking connections at the beginning of shutting down.
+    // This needs to happen before unloading the level because the network connection might queue
+    // references to entities and assets that could prevent cleanup from happening during the level unload.
+    if (const auto console = AZ::Interface<AZ::IConsole>::Get())
+    {
+        console->PerformCommand("disconnect");
+    }
+
+    // On shutdown, we need to start by unloading the level spawnable and processing the spawnable queue
+    // to clean up any remaining references. Otherwise, we'll get lots of errors on shutdown due to assets still
+    // being in use as various subsystems get shut down. By the time the spawnable system shuts down, it will try
+    // to clean up the assets, but the asset handlers for those assets will also be deregistered and destroyed by then,
+    // causing even more errors.
+    // By unloading the level before shutting down any subsystems, we can avoid the cascade of errors.
+    ILevelSystem* levelSystem = GetILevelSystem();
+    if (levelSystem)
+    {
+        levelSystem->UnloadLevel();
+        if (auto spawnableInterface = AzFramework::RootSpawnableInterface::Get(); spawnableInterface)
+        {
+            spawnableInterface->ProcessSpawnableQueueUntilEmpty();
+        }
+    }
```

Comment:
Can't we have the [SpawnableLevelSystem destructor](https://github.com/o3de/o3de/blob/development/Code/Legacy/CrySystem/LevelSystem/SpawnableLevelSystem.cpp#L109-L112) do this instead?

Reply:
That would be really nice but it looks like it would take a larger amount of system untangling / bugfixing. In trying it, I get a crash on the level unload from ILyShine being null, which appears to be because of the work it does in OnCrySystemShutdown: https://github.com/o3de/o3de/blob/73ea5cae2d6f8ca2243e786ac8c0f5414af579c1/Gems/LyShine/Code/Source/LyShineSystemComponent.cpp#L405

I'm hesitant to push the level destructor above that notification, since destroying the system _is_ a part of shutting down. It would also then cause me to fix that LyShine code where it calls GetILevelSystem()->RemoveListener() to null guard it, and might have some other knock-off effects. I suspect it will likely start causing other small pieces to unravel and require fixing too.

For now, I'd propose we leave this here as a start. 

If/when anyone feels inclined to take another pass at improving our shutdown code, hopefully we could make the disconnect logic work just through standard Multiplayer shutdown without requiring it to happen via console command here, and hopefully we could break apart these CrySystem bits a little further so that the level system, CryFont, and LyShine can all just shut down normally in their own systems without implicit ordering dependencies.

Reply:
That is good start and good with me.

It still sucks that despite being a gem LyShine is still entangled with [CrySystem](https://github.com/o3de/o3de/blob/development/Gems/LyShine/Code/Source/LyShineSystemComponent.cpp#L383-L414)